internal/app/connectconformance/server_runner.go

```diff
@@ -30,140 +30,179 @@ import (
 	"google.golang.org/protobuf/types/known/anypb"
 )
 
-// populateExpectedRequests populates the requests we expected to get back from the server
-// by examining the requests we sent.
-func populateExpectedRequests(testCase *conformancev1alpha1.TestCase) error {
-	// If an expected response was already provided, return and use that.
-	// This allows for overriding this function with explicit values in the yaml file.
-	if testCase.ExpectedResponse != nil {
+// populates the expected response for a unary test case.
+func populateExpectedUnaryResponse(testCase *conformancev2.TestCase) error {
+	req := testCase.Request.RequestMessages[0]
+	// First, find the response definition that the client instructed the server to return
+	concreteReq, err := req.UnmarshalNew()
+	if err != nil {
+		return err
+	}
+	type unaryResponseDefiner interface {
+		GetResponseDefinition() *conformancev2.UnaryResponseDefinition
+	}
+
+	definer, ok := concreteReq.(unaryResponseDefiner)
+	if !ok {
+		return fmt.Errorf("%T is not a unary test case", concreteReq)
+	}
+
+	// TODO - Need to define this better in the protos and tests as to how services should
+	// behave if no responses are specified. The behavior right now differs for unary vs. streaming
+	// If no responses are specified for unary, the service will still return a response with the
+	// request information inside (but none of the response information since it wasn't provided)
+	// But streaming endpoints don't return a single response and instead return responses via sending
+	// on a stream. But if no responses are specified in the request, the streams don't send anything outbound
+	// so there's no way to relay this to a client. So right now, streaming endpoints simply expect an empty
+	// ClientResponseResult if no response definition is provided
```

Comment:
Interesting point. TBH, we can probably get away with doing nothing -- if the response definition says to send back no messages, then the request doesn't get echoed back. And that's probably okay since we'll have coverage of the request handling and it being echoed back from so many other test cases where there is 1+ response messages.